libs/service-portal/assets/src/utils/parseFileToMileage.ts

```
@@ -0,0 +1,125 @@
+
+const parseCsvString = (chunk: string): Promise<string[][]> => {
+  return new Promise((resolve, reject) => {
+    const records: string[][] = []
+
+    const parser = parse({
+      skip_empty_lines: true,
+    })
+
+    parser.on('readable', () => {
+      let record
+      while ((record = parser.read()) !== null) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)